src/infrastructure/jira/jira-service.ts

```diff
@@ -135,9 +134,8 @@ class JiraService {
 				connectInstallation,
 			);
 
-			const storedAttachedDesignUrlIssuePropertyValues = JSON.parse(
-				ensureString(response.value),
-			) as AttachedDesignUrlV2IssuePropertyValue[];
+			const storedAttachedDesignUrlIssuePropertyValues =
```

Comment:
**suggestion:** Taking into account that our Connect App does not have an exclusive ownership over these Issue Properties, it can contain any value. Therefore, it can be safer to validate the schema of their values.

Here is what [Jira docs](https://developer.atlassian.com/cloud/jira/platform/jira-entity-properties) say:

> Entity properties can be modified by all users that have permission to edit the entity. Entity properties can be modified by all apps in the system and exist in a global namespace.

> Be defensive when retrieving app properties, and don't assume data consistency (arbitrary keys may be modified or deleted by users).

Reply:
Added validation